src/components/Contacts/ContactDetails/ContactDontationsTab/PartnershipInfo/EditPartnershipInfoModal/EditPartnershipInfoModal.tsx

```diff
@@ -0,0 +1,431 @@
+  padding: theme.spacing(0, 6),
+  margin: theme.spacing(2, 0),
+}));
+
+const LoadingIndicator = styled(CircularProgress)(({ theme }) => ({
+  margin: theme.spacing(0, 1, 0, 0),
+}));
+
+const CheckboxLabel = styled(FormControlLabel)({
+  margin: 'none',
+});
+
+interface EditPartnershipInfoModalProps {
+  contact: ContactDonorAccountsFragment;
+  handleClose: () => void;
+  isOpen: boolean;
+}
+
+export const EditPartnershipInfoModal: React.FC<EditPartnershipInfoModalProps> = ({
+  contact,
+  handleClose,
+  isOpen,
+}) => {
+  const { t } = useTranslation();
+  const accountListId = useAccountListId();
+  const constants = useApiConstants();
+
+  const { enqueueSnackbar } = useSnackbar();
+  const [
+    updateContactPartnership,
+    { loading: updating },
+  ] = useUpdateContactPartnershipMutation();
+  const pledgeCurrencies = constants?.pledgeCurrencies;
+  const method = contact.lastDonation
+    ? contact.lastDonation.paymentMethod
+    : t('None');
+
+  const lastGift = contact.lastDonation
+    ? `${currencyFormat(
+        contact.lastDonation.amount.amount,
+        contact.lastDonation.amount.currency,
+      )}`
+    : '';
+
+  const contactPartnershipSchema: yup.SchemaOf<
+    Pick<
+      ContactUpdateInput,
+      | 'id'
+      | 'status'
+      | 'pledgeAmount'
+      | 'pledgeFrequency'
+      | 'pledgeCurrency'
+      | 'pledgeReceived'
+      | 'pledgeStartDate'
+      | 'nextAsk'
+      | 'noAppeals'
+    >
+  > = yup.object({
+    id: yup.string().required(),
+    status: yup.mixed<StatusEnum>().oneOf(Object.values(StatusEnum)).nullable(),
+    pledgeAmount: yup.number().moreThan(-1).nullable(),
+    pledgeStartDate: yup.string().nullable(),
+    pledgeReceived: yup.boolean().default(false).nullable(),
+    pledgeCurrency: yup.string().nullable(),
+    nextAsk: yup.string().nullable(),
+    noAppeals: yup.boolean().default(false).nullable(),
+    pledgeFrequency: yup.mixed<PledgeFrequencyEnum>().nullable(),
+  });
+
+  const onSubmit = async (
+    attributes: Pick<
+      ContactUpdateInput,
+      | 'id'
+      | 'status'
+      | 'pledgeAmount'
+      | 'pledgeFrequency'
+      | 'pledgeCurrency'
+      | 'pledgeReceived'
+      | 'pledgeStartDate'
+      | 'nextAsk'
+      | 'noAppeals'
+    >,
+  ) => {
+    console.log(attributes.nextAsk);
+    await updateContactPartnership({
+      variables: {
+        accountListId: accountListId ?? '',
+        attributes,
+      },
+    });
+
+    enqueueSnackbar(t('Partnership information updated successfully.'), {
+      variant: 'success',
+    });
+    handleClose();
+  };
+
+  const updateStatus = (
+    status: StatusEnum,
+    setFieldValue: (name: string, value: StatusEnum | number | null) => void,
+  ) => {
+    if (status !== StatusEnum.PartnerFinancial) {
+      setFieldValue('status', status);
+      setFieldValue('pledgeAmount', 0);
+      setFieldValue('pledgeFrequency', null);
+    } else {
+      setFieldValue('status', status);
+    }
+  };
```

Comment:
The expected behavior here is that there should only be an amount and frequency when the ```status``` is ```PartnerFinancial```. So this clears the input fields and I disable them from being edited and show a hover icon explaining the behavior, but only when the new status is not ```PartnerFinancial```. I wanted to set ``` pledgeAmount``` to ```null``` but since the input type is ```type="number"``` it causes some really strange behavior if you set the value to ```null```, so I just settled for ```0```.